vector/src/main/java/im/vector/app/features/spaces/manage/SpaceManageActivity.kt

```diff
@@ -72,14 +95,33 @@ class SpaceManageActivity : VectorBaseActivity<ActivitySimpleBinding>(), SpaceMa
 
         val args = intent?.getParcelableExtra<SpaceManageArgs>(MvRx.KEY_ARG)
         if (isFirstCreation()) {
-            val simpleName = SpaceAddRoomFragment::class.java.simpleName
-            if (supportFragmentManager.findFragmentByTag(simpleName) == null) {
-                supportFragmentManager.commitTransaction {
-                    replace(R.id.simpleFragmentContainer,
-                            SpaceAddRoomFragment::class.java,
-                            Bundle().apply { this.putParcelable(MvRx.KEY_ARG, args) },
-                            simpleName
-                    )
+            withState(sharedViewModel) {
+                when (it.manageType) {
+                    ManageType.AddRooms -> {
+                        val simpleName = SpaceAddRoomFragment::class.java.simpleName
+                        if (supportFragmentManager.findFragmentByTag(simpleName) == null) {
+                            supportFragmentManager.commitTransaction {
+                                replace(R.id.simpleFragmentContainer,
+                                        SpaceAddRoomFragment::class.java,
+                                        Bundle().apply { this.putParcelable(MvRx.KEY_ARG, args) },
+                                        simpleName
+                                )
+                            }
+                        }
+                    }
+                    ManageType.Settings -> {
+                        val simpleName = SpaceSettingsFragment::class.java.simpleName
+                        if (supportFragmentManager.findFragmentByTag(simpleName) == null) {
+                            supportFragmentManager.commitTransaction {
+                                replace(R.id.simpleFragmentContainer,
+                                        SpaceSettingsFragment::class.java,
+                                        Bundle().apply { this.putParcelable(MvRx.KEY_ARG, RoomProfileArgs(args?.spaceId ?: "")) },
```

Comment:
Do not do anything if `args?.spaceId` is null?

Reply:
done